src/core/geometry/qgscurvepolygon.cpp

```diff
@@ -435,6 +442,52 @@ QgsAbstractGeometry *QgsCurvePolygon::boundary() const
   }
 }
 
+QgsCurvePolygon *QgsCurvePolygon::asGridified( double hSpacing, double vSpacing, double dSpacing, double mSpacing,
+    double tolerance, SegmentationToleranceType toleranceType ) const
+{
+  if ( !mExteriorRing )
+    return nullptr;
+
+  QgsPolygonV2 *polygon;
```

Comment:
polygon should be a unique_ptr

Reply:
`polygon`is the result. The result of snap to grid on any type of `QgsCurvePolygon` is always `QgsPolygon` or a subclass. But I can't return a `QgsPolygon *`. Maybe I should change the name of the variable to make it more explicit.

Reply:
I just mean it shouldn't be a raw pointer - there's too much risk of a leak if early return paths are added later. In general I think we should be moving toward the c++>=11 practice of never calling "new" without storing the result in a shared/unique_ptr (with the exception of times when we're forced to by the Qt API).

Reply:
Forgot that `std::unique_ptr` have `release()`.